genesis/engine/solvers/sf_solver.py

```diff
@@ -97,12 +97,13 @@ def reset_swap(self):
     @ti.kernel
     def pressure_jacobi(self, pf: ti.template(), new_pf: ti.template()):
         for I in ti.grouped(ti.ndrange(*self.res)):
-            pl = pf[self.compute_location(*I, -1, 0, 0)]
-            pr = pf[self.compute_location(*I, 1, 0, 0)]
-            pb = pf[self.compute_location(*I, 0, -1, 0)]
-            pt = pf[self.compute_location(*I, 0, 1, 0)]
-            pp = pf[self.compute_location(*I, 0, 0, -1)]
-            pq = pf[self.compute_location(*I, 0, 0, 1)]
+            u, v, w = I
```

Comment:
Just do `for u, v, w in ti.ndrange(*self.res):`

Reply:
addressed in decafcc4